gtsam/linear/VectorValues.h

```diff
@@ -197,7 +204,7 @@ namespace gtsam {
      *  and an iterator to the existing value is returned, along with 'false'.  If the value did not
      *  exist, it is inserted and an iterator pointing to the new element, along with 'true', is
      *  returned. */
-    std::pair<iterator, bool> tryInsert(Key j, const Vector& value) {
+    inline std::pair<iterator, bool> tryInsert(Key j, const Vector& value) {
```

Comment:
I think `tryInsert` is now obsolete? We should put it inside DEPRECATED block, and its comments should migrate to emplace, referencing that this is standard emplace behavior anyway :-)

Reply:
![image](https://user-images.githubusercontent.com/5358374/83589977-cb823280-a522-11ea-839c-1c05230b3d26.png)

Seems we still have a lot of users :) I propose dealing with this in a separate PR.

Reply:
OK. And I propose that in that PR we (1) examine all those cases to see if we can save on a malloc there as well, and (2) deprecate all but emplace for non-const access. We should be opinionated about how to deal w VectorValues. It will be mostly internal use anyway.